src/main/java/cz/smarteon/loxone/LoxoneWebSocket.java

```diff
@@ -305,6 +305,8 @@ private void waitForRetry() {
     }
 
 
+    // TODO Contains potential race condition when response to the sent command is received faster than command is added to queue
+    // however, the probability it happens with real miniserver is very low
```

Comment:
Couldn`t be easily solved by switching the order of commands?
```
if (! KEEP_ALIVE.getCommand().equals(command.getCommand())) {
    commands.add(command);
}
webSocketClient.send(command.getCommand());
```

Reply:
It couldn't be solved easily. If you switch order of statements and `websocket.send()` won't process correctly the commands list have inconsistent state. The proper solution is to make the section critical.

Reply:
Adding another `ReadWriteLock` and guarding all the access to the commands is most correct and relatively easy to implement. The down side is the code readability and slight perf decrease. Considering in reality this race condition could in fact happen only in case the network round trip and miniserver processing will happen in less than few dozens of milliseconds it's probably not worth to add critical section handling here.

Reply:
ok, now I see it, thx for explanation.